xblock/runtime.py

```diff
@@ -1267,17 +1268,42 @@ def __getattr__(self, name):
         return getattr(self._translations, name)
 
     STRFTIME_FORMATS = {
-        "SHORT_DATE_FORMAT": "%b %d, %Y",
-        "LONG_DATE_FORMAT": "%A, %B %d, %Y",
-        "TIME_FORMAT": "%I:%M:%S %p",
-        "DATE_TIME_FORMAT": "%b %d, %Y at %H:%M",
+        u"SHORT_DATE_FORMAT": u"%b %d, %Y",
+        u"LONG_DATE_FORMAT": u"%A, %B %d, %Y",
+        u"TIME_FORMAT": u"%I:%M:%S %p",
+        u"DATE_TIME_FORMAT": u"%b %d, %Y at %H:%M",
     }
 
     def strftime(self, dtime, format):      # pylint: disable=redefined-builtin
         """
         Locale-aware strftime, with format short-cuts.
         """
         format = self.STRFTIME_FORMATS.get(format + "_FORMAT", format)
-        if isinstance(format, unicode):
-            format = format.encode("utf8")
-        return dtime.strftime(format).decode("utf8")
+        if future.utils.PY2 and isinstance(format, str):
+            format = format.encode("utf-8")
+        timestring = dtime.strftime(format)
+        if future.utils.PY2:
+            timestring = timestring.decode('utf-8')
```

Comment:
I feel like all these custom chunks that branch on `future.utils.PY2` should be pulled out into an edx-future library.  I think we'll need them in a number of places.